### PR TITLE
ui, presets: adjust for new cvars and cvar types

### PR DESCRIPTION
**Pull request by illwieckz**
The graphics menu now list light mapping and vertex lighting and makes use of the r_lightMode cvar.

The lowest graphics preset now uses vertex lighting.

See:

- https://github.com/DaemonEngine/Daemon/pull/992

### CODE REVIEW

**Comment by illwieckz** [comment]
Related engine PR has been merged:

- https://github.com/DaemonEngine/Daemon/pull/992

It means that without this patch the Light option in UI is currently doing nothing.

**Comment by illwieckz** [comment]
> I guess the old one shouldn't be deleted from the presets until 0.55 is released?

There was no vertex lighting implemented before, and all presets were enabling light mapping.

Setting `r_lightMode` on 0.54.1 will do nothing.

But the best is probably to wait for the release to merge this (if we do a `0.54.2` for portal and lightmap fixes), or merge to `for-0.55.0/sync`. In all cases the _Lights_ menu will be broken on 0.54.1 once this is merged.

**Comment by illwieckz** [comment]
The branch is now targetting`for-0.55.0/sync`, I added more commits for new cvars or new cvar types.

**Review by DolceTriade — APPROVED** [review]

**Comment by illwieckz** [comment]
I added presets and option for `r_highPrecisionRendering`, it is preferred to disable it on lower-end devices, and may even workaround driver bugs, see:

- https://github.com/DaemonEngine/Daemon/issues/1232

File: pkg/unvanquished_src.dpkdir/ui/options_graphics.rml
Comment (illwieckz) [reply pkg/unvanquished_src.dpkdir/ui/options_graphics.rml:113]: In fact the need was to avoid building the non-static dynamic light renderer.

I have a better solution for that:

https://github.com/DaemonEngine/Daemon/pull/1234

Comment (illwieckz) [reply pkg/unvanquished_src.dpkdir/ui/options_graphics.rml]: The new cvar r_realtimeLighting` is now used and `r_dynamicLight` and `r_staticLight` are not modified anymore.

File: pkg/unvanquished_src.dpkdir/ui/options_graphics.rml
Comment (illwieckz) [reply pkg/unvanquished_src.dpkdir/ui/options_graphics.rml]: Deleted.

File: pkg/unvanquished_src.dpkdir/presets/graphics/high.cfg
Comment (illwieckz) [reply pkg/unvanquished_src.dpkdir/presets/graphics/high.cfg:19]: Yeah but why not? My idea is that we progressively migrate cvars to new style ones and presets when we can, and then once there is no legacy cvar anymore, we will modify he RmlUi checkbox to switch between on/off instead of 0/1. This change is pure bikeshedding, is not required, but why not. When I add new-style boolean cvars to presets I already use on/off, so I prefer to unify them.

**Comment by illwieckz** [comment]
I imported the commit to not modify `r_halfLambertLighting` (that used the same value in all presets) from #3094:

- from https://github.com/Unvanquished/Unvanquished/pull/3094

It should also be noted that half lambert lighting is only used when deluxe mapping is enabled, so even the idea of disabling it in a low preset doesn't make sense because it is already disabled  in such preset by disabling deluxe mapping to begin with.

**Comment by illwieckz** [comment]
I removed the `r_clear` cvar setting, as the engine now know how to clear when `r_fastsky` is enabled, without extra cvar.

**Comment by illwieckz** [comment]
I added two `reset` lines for old cvars that can now be set to default (`r_dynamicLight` and `r_clear`).

The `r_halfLambertLighting` cvar previously being set to the same version in all presets was probably a bad attempt at doing the same for this cvar in the past. I can add an explicit reset line for it.

The idea to ship such reset lines in presets is that old user profiles with such previously-used cvar can be fixed by the user by selecting a preset.

For example if one someone used a low preset disabling `r_dynamicLight` in the past, then later upgrade his hardware and use a high preset, the `r_dynamicLight` cvar should be reset despite no preset modifies it anymore, otherwise the specific dynamic light kind of realtime lights will not be enabled while realtime lights will be enabled.

File: pkg/unvanquished_src.dpkdir/ui/options_graphics.rml
Comment (illwieckz) [reply pkg/unvanquished_src.dpkdir/ui/options_graphics.rml]: Done.

File: pkg/unvanquished_src.dpkdir/presets/graphics/high.cfg
Comment (slipher) [reply pkg/unvanquished_src.dpkdir/presets/graphics/high.cfg:19]: Well 0/1 is just more reliable, it always works. Old-style cvars, new booleans, or things that used to be booleans and got changed to something with 3 options, 0/1 works with any. Making a word format the default serialization is a mistake if you ask me.

**Review by slipher — COMMENTED** [review]
LGTM

**Comment by illwieckz** [comment]
@slipher I added two commits as I found two other cvars to take care about:

- use new `r_imageFitScreen` in both menu and presets,
- replace now unused `r_picmip` by better `r_imageMaxDimension` in menu (presets already did that).

File: pkg/unvanquished_src.dpkdir/ui/options_graphics.rml
Comment (slipher) [inline pkg/unvanquished_src.dpkdir/ui/options_graphics.rml:144]: ISTM this one should be sorted as higher than ultra.

**Comment by slipher** [comment]
If we merge the r_dynamicLight rename in https://github.com/DaemonEngine/Daemon/pull/1341 we can get rid of the `reset r_dynamicLight`.

File: pkg/unvanquished_src.dpkdir/ui/options_graphics.rml
Comment (illwieckz) [reply pkg/unvanquished_src.dpkdir/ui/options_graphics.rml]: Done.

**Comment by illwieckz** [comment]
> If we merge the r_dynamicLight rename in [DaemonEngine/Daemon#1341](https://github.com/DaemonEngine/Daemon/pull/1341) we can get rid of the `reset r_dynamicLight`.

Done.

**Comment by illwieckz** [comment]
I added a slight modification: lowest presets now use `r_imageMaxDimensions` set to `16`. Lowest is meant to be “lowest”, and I've always used `16` on the lowest hardware range for the last months and it does help, on both computational need and on low memory pressure, while keeping enough details in textures so the world still looks and feels textured (unlike `1`). Actually lowest is for things like Intel GMA 3, Intel GMA 4, VC4 and very low end devices like that. Even the early R300 cards can now do “low” thanks to all the improvements we done (float vertexes, low quality models, particle optimizations etc.)

**Comment by slipher** [comment]
LGTM

**Pull request merged**
